File: tasks/task12_1/src/Point.py
from math import hypot  

class Point: pass # avoid error on line 13

class Point:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def distance(self) -> float:
        return hypot(self.x, self.y)

    def distanceTo(self, p: Point) -> float:
        return hypot((self.x - p.x), (self.y - p.y))



def main(): 
    print("enter x coord:")
    inputX = float(input())
    print("enter y coord:")
    inputY = float(input())
    p = Point(inputX, inputY)
    print(f"Dist from  origin is {p.distance()}")
    print(f"Dist from  4.5,7 is {p.distanceTo(Point(4.5,7.0))}")



if __name__ == "__main__":
    main()
    

File: tasks/task12_1/src/Point.kt
import kotlin.math.hypot

class Point(var x: Double, var y: Double) {
    fun distance() = hypot(x, y)

    fun distanceTo(p: Point) = hypot((x - p.x), (y - p.y))
}

fun main() {
    println("enter x coord:")
    val inputX = readln().toDouble()
    println("enter y coord:")
    val inputY = readln().toDouble()
    val p = Point(inputX, inputY)
    println("Dist from  origin is ${p.distance()}")
    println("Dist from  4.5,7 is ${p.distanceTo(Point(4.5,7.0))}")
}
